accounts/src/main/java/eu/kevin/accounts/linkingsession/AccountLinkingSession.kt

```diff
@@ -47,15 +53,33 @@ internal class AccountLinkingSession(
 
     fun beginFlow(listener: AccountLinkingSessionListener?) {
         sessionListener = listener
-        if (currentFlowIndex == -1) {
-            sessionData = sessionData.copy(
-                selectedCountry = configuration.preselectedCountry,
-                selectedBank = configuration.preselectedBank
-            )
-        }
-        initializeFlow()
-        if (currentFlowIndex == -1) {
-            GlobalRouter.pushFragment(getFlowFragment(0))
+
+        lifecycleOwner.lifecycleScope.launch(Dispatchers.IO) {
+            val selectedBank = if (configuration.preselectedBank != null) {
+                withContext(Dispatchers.Main) {
+                    sessionListener?.showLoading(true)
+                }
+                val apiBanks = accountsClient.getSupportedBanks(configuration.state, configuration.preselectedCountry)
```

Comment:
this will crash as the exception is not handled. are you sure network requests should be made here?

Reply:
I need to get selected bank object in case bank selection window will be skipped